### PR TITLE
adds observedfun for DAEFunction

### PR DESCRIPTION
**Pull request by ValentinKaisermayer**

### CODE REVIEW

**Comment by ValentinKaisermayer** [comment]
closes #1589, #1436

**Comment by ChrisRackauckas** [comment]
Run the formatter

**Comment by ValentinKaisermayer** [comment]
The stuff in MTKSL is no longer broken

File: src/systems/diffeqs/abstractodesystem.jl [hunk 1]
```diff
@@ -509,6 +509,16 @@ function DiffEqBase.DAEFunction{iip}(sys::AbstractODESystem, dvs = states(sys),
         _jac = nothing
     end
 
+    obs = observed(sys)
+    observedfun = let sys = sys, dict = Dict()
+        function generated_observed(obsvar, u, p, t)
```
Comment (YingboMa) [inline src/systems/diffeqs/abstractodesystem.jl:514]: This would error if the observed variable depends on derivative variables.

**Comment by YingboMa** [comment]
Thanks for be PR! But it doesn't fully solve the problem though.

**Comment by ValentinKaisermayer** [comment]
You are welcome. 
Maybe a (broken) test should be added for such a general case?

**Pull request merged**